addons/binding/org.openhab.binding.nest/ESH-INF/thing/channels.xml

```diff
@@ -0,0 +1,319 @@
+    </channel-type>
+
+    <channel-type id="PublicShareEnabled" advanced="true">
+        <item-type>Switch</item-type>
+        <label>Public Share Enabled</label>
+        <description>If the public sharing of this camera is enabled.</description>
+    </channel-type>
+
+    <channel-type id="Streaming">
+        <item-type>Switch</item-type>
+        <label>Streaming</label>
+        <description>If the camera is currently streaming.</description>
+    </channel-type>
+
+    <channel-type id="WebUrl">
+        <item-type>String</item-type>
+        <label>Web Url</label>
+        <description>The web url for the camera, allows you to see the camera in a web page.</description>
+    </channel-type>
+
+    <channel-type id="PublicShareUrl">
+        <item-type>String</item-type>
+        <label>Public Share Url</label>
+        <description>The url for the camera which is open for a public share.</description>
+    </channel-type>
+
+    <channel-type id="SnapshotUrl" advanced="true">
+        <item-type>String</item-type>
+        <label>Snapshot Url</label>
+        <description>The url for the camera showing a snapshot of the camera.</description>
+    </channel-type>
+
+    <channel-type id="AppUrl" advanced="true">
+        <item-type>String</item-type>
+        <label>App Url</label>
+        <description>The app url for the camera, allows you to see the camera in an app.</description>
+    </channel-type>
+
+    <!-- smoke detector -->
+    <!-- TODO could / should we use a real color item instead -->
+    <channel-type id="UiColorState" advanced="true">
+        <item-type>String</item-type>
+        <label>Mode</label>
+        <description>Current color state of the protect</description>
+        <state readOnly="true">
+            <options>
+                <option value="GRAY">gray</option>
+                <option value="GREEN">green</option>
+                <option value="YELLOW">yellow</option>
+                <option value="RED">red</option>
+            </options>
+        </state>
+    </channel-type>
+
+    <channel-type id="CoAlarmState">
+        <item-type>String</item-type>
+        <label>CO Alarm State</label>
+        <description>Carbon Monoxide alarm state</description>
+        <state readOnly="true">
+            <options>
+                <option value="OK">ok</option>
+                <option value="EMERGENCY">emergency</option>
+                <option value="WARNING">warning</option>
+            </options>
+        </state>
+    </channel-type>
+
+    <channel-type id="SmokeAlarmState">
+        <item-type>String</item-type>
+        <label>Smoke Alarm State</label>
+        <description>Smoke Alarm state</description>
+        <state readOnly="true">
+            <options>
+                <option value="OK">ok</option>
+                <option value="EMERGENCY">emergency</option>
+                <option value="WARNING">warning</option>
+            </options>
+        </state>
+    </channel-type>
+
```

Comment:
This was already done, this one can be removed, maybe check for other unused ones